realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -930,22 +956,36 @@ private void emitCreateOrUpdateUsingJsonObject(JavaWriter writer) throws IOExcep
         if (!metadata.hasPrimaryKey()) {
             writer.emitStatement("%s obj = realm.createObject(%s.class)", className, className);
         } else {
+            // Indentations for emitStatement, beginControlFlow, and endControlFlow is made to follow
```

Comment:
I would remove this statement here. It should probably be in the Javadoc for the entire class.